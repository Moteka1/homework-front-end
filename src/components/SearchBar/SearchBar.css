input:focus {
  outline-width: 0;
}

.searchTerm {
  width: 70%;
  height: 40px;
  margin: 20px 0 20px 20px;
  padding-left: 1em;
  border-radius: 8px;
}

.searchbar-button {
  background-color: #ea4640;
  border-radius: 8px;
  height: 46px;
  color: white;
}

.searchbar-button:hover,
.searchbar-button:focus {
  background: #0053ba;
}

.searchbar-button:focus {
  outline-offset: -4px;
}

.searchbar-button:active {
  transform: scale(0.99);
}

@media only screen and (max-width: 620px) {
  /* For mobile phones: */
  .searchTerm,
  .searchbar-button {
    margin: 0;
  }
}
