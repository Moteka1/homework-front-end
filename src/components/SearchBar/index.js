import React, { Component } from 'react';
import './SearchBar.css';

export default class SearchBar extends Component {
  // instead of constructor and super(props)
  state = {
    searchTerm: ''
  };

  // use arrow function instead of bind to give "this" the correct context
  handleChange = evt => {
    this.setState({
      [evt.target.name]: evt.target.value
    });
  };

  // handle submitting the search bar form
  // send term to parent (Homepage.js) in order to make API call
  handleSubmit = evt => {
    evt.preventDefault();
    this.props.search(this.state.searchTerm);
    this.setState({ searchTerm: '' });
  };

  render() {
    return (
      <div className="searchbar-container">
        <div className="searchbar-content">
          <form onSubmit={this.handleSubmit}>
            <label htmlFor="search">
              <input
                onChange={this.handleChange}
                type="text"
                id="search"
                name="searchTerm"
                className="searchTerm"
                value={this.state.searchTerm}
                placeholder="Search for Gif"
              />
            </label>
            <input type="submit" value="Search" className="searchbar-button" />
          </form>
        </div>
      </div>
    );
  }
  // need to have a button and a search input/form
  // this value will
}
