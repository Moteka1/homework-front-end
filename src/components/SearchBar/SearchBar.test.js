import React from 'react';
import SearchBar from './index.js';
import renderer from 'react-test-renderer';

describe('<SearchBar />', () => {
  it('matches the snapshot', () => {
    var tree = renderer.create(<SearchBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
