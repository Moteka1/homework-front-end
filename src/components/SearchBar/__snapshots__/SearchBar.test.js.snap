// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> matches the snapshot 1`] = `
<div
  className="searchbar-container"
>
  <div
    className="searchbar-content"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="search"
      >
        <input
          className="searchTerm"
          id="search"
          name="searchTerm"
          onChange={[Function]}
          placeholder="Search for Gif"
          type="text"
          value=""
        />
      </label>
      <input
        className="searchbar-button"
        type="submit"
        value="Search"
      />
    </form>
  </div>
</div>
`;
